include/ensmallen_bits/pso/pso.hpp

```diff
@@ -0,0 +1,138 @@
+/**
+ * @file pso.hpp
+ * @author Chintan Soni
+ * @author Suryoday Basak
+ *
+ * Particle swarm optimization.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_PSO_PSO_HPP
+#define ENSMALLEN_PSO_PSO_HPP
+
+#include "update_policies/lbest_update.hpp"
+#include "init_policies/default_init.hpp"
+#include "init_policies/constrained_init.hpp"
+#include <iostream>
+
+namespace ens {
+
+/**
+ * EXTREMELY DETAILED DESCRIPTION OF THE WORKING OF PSO.
+ *
+ * For Particle Swarm Optimization to work, a FunctionType template parameter is
+ * required.
+ * This class must implement the following function:
+ *
+ *   double Evaluate(const arma::mat& coordinates);
+ *   void Gradient(const arma::mat& coordinates,
+ *                 arma::mat& gradient);
+ */
```

Comment:
Would be great if you could add a reference to the function defenitions:
https://github.com/mlpack/ensmallen/blob/8d9682ca54483fb0601db8b48b17839e206149eb/include/ensmallen_bits/ftml/ftml.hpp#L41-L43

is just one example.

Reply:
Could you please clarify this? What I'm seeing is that you want me to add references to ftml.hpp?

Reply:
Sorry for the confusion, I was talking about adding a reference to the function type documentation: 
```
For more details, see 
  * the documentation on function types include with this distribution or on the 
 * ensmallen website. 
```